iceoryx_posh/include/iceoryx_posh/gateway/channel.hpp

```diff
@@ -51,44 +49,44 @@ enum class ChannelError : uint8_t
 /// cleaning them up when the channel is discarded.
 /// This can be achieved via the Channel::create method.
 ///
-template <typename IceoryxTerminal, typename DDSTerminal>
+template <typename IceoryxTerminal, typename ExternalTerminal>
 class Channel
 {
     using IceoryxTerminalPtr = std::shared_ptr<IceoryxTerminal>;
-    using IceoryxTerminalPool = iox::cxx::ObjectPool<IceoryxTerminal, MAX_CHANNEL_NUMBER>;
-    using DDSTerminalPtr = std::shared_ptr<DDSTerminal>;
-    using DDSTerminalPool = iox::cxx::ObjectPool<DDSTerminal, MAX_CHANNEL_NUMBER>;
+    using IceoryxTerminalPool = cxx::ObjectPool<IceoryxTerminal, MAX_CHANNEL_NUMBER>;
+    using ExternalTerminalPtr = std::shared_ptr<ExternalTerminal>;
+    using ExternalTerminalPool = cxx::ObjectPool<ExternalTerminal, MAX_CHANNEL_NUMBER>;
 
   public:
-    Channel(const iox::capro::ServiceDescription& service,
+    Channel(const capro::ServiceDescription& service,
             const IceoryxTerminalPtr iceoryxTerminal,
-            const DDSTerminalPtr ddsTerminal) noexcept;
+            const ExternalTerminalPtr externalTerminal) noexcept;
 
-    constexpr bool operator==(const Channel<IceoryxTerminal, DDSTerminal>& rhs) const noexcept;
+    constexpr bool operator==(const Channel<IceoryxTerminal, ExternalTerminal>& rhs) const noexcept;
```

Comment:
To use the constexpr comparision operator during compile time we need a constexpr constructor. Is this at the moment possible? 
I think we need then also a constexpr ServiceDescription ctor and with that a constexpr cxx::string constructor.

I would definitely leave the comparision operator constexpr but I think we need to raise the awareness that we should make everything constexpr if possible.

Reply:
I think all of that is possible in theory ... I won't go deeper than making the channel constructor constexpr in this PR though.